[Chore #163020832] Refactored user interface(UI) and API-endpoints controller

abidex4yemi opened the pull request:
#### What does this PR do?
This allows code base to be scalable and maintainable
for future reviews and adding of a new feature to the code
base. Also makes the website to be responsive on any device. 

#### Description of Task to be completed?
* Have the following UI page element responsive

` Home page: question_feeds_action container should be responsive and align within the parent container`

`upcoming meetups page: Attend meetup button should be clickable and also link to the sign-in page.`

`Dashboard page: dashboard sidebar height should be 100% of the viewport height in respective of the viewport.`

`Dashboard page: dashboard sidebar content should be well aligned within their parent container`

* Have the following API-endpoints refactored 

` MeetupController: MeetupController object should be converted to a single class file`

` MeetupController: all method with async without await pears should be removed.`

`MeetupController: all method name should be prepended with the static keyword for usability without instantiation`

` QuestionController: QuestionController object should be converted to a single class file`

` QuestionController: all method with async without await pears should be removed.`

` QuestionController: all method name should be prepended with the static keyword for usability without instantiation`

`Helper files: all separate function files should move to a single class file`

` output: all console.log method should be removed from the code base`

#### How should this be manually tested?
* By visiting this link below the UI can be viewed and tested.
   Home page [home][1]
   Dashboard page [dashboard][2]

  [1]:  https://abidex4yemi.github.io/meet-up-questioner/index.html
  [2]:  https://abidex4yemi.github.io/meet-up-questioner/admin-dashboard.html

* By visiting the GitHub repo link below the controller files can view for change being made
[meet-up-questioner] (https://github.com/abidex4yemi/meet-up-questioner).


#### What are the relevant pivotal tracker stories?
#163020832

abidex4yemi committed:
[Chore #163020832] Refactored UI and Api controller

This allows code base to be scale able and maintain able
for future reviews and adding of new feature to the code
base.

This commit will update exiting code base with new refactored
changes from this chore branch and makes more improvement
to the application user interface and api-endpoints.
Listed below are the files and folder being refactored:
	* Api-endpoints
		- Removed and replaced async with static keyword
		- Moved all controller methods  from object to a single
		class
		- Moved all helper methods to a single class
		- removed all console.log from code base
	* Front-end (UI)
		- Fixed dashboard  min-height issue on larger screens
		- Fixed dashboard sidebar container alignment issues
		- Fixed alignment issues on index page
		  (question-feeds-actions),upvote,downvote,comments
		- Fixed upcoming meetup page issues (attend meetup button)
		  to be click able and link to sign in.

Resolves: https://www.pivotaltracker.com/n/projects/2232110 [#163020832]

saheedt commented:
Looks good, you can go ahead to merge this.

saheedt commented:
I just checked your `gh-pages` which i believe came from this changes and saw not all your pages are responsive.
<img width="1431" alt="screenshot 2019-01-08 at 6 10 12 pm" src="https://user-images.githubusercontent.com/8380409/50847117-0ead9580-1371-11e9-9a85-0401ca8ac322.png">
Please make use of the `chrome devtools` to simulate different screen sizes.

abidex4yemi committed:
Updated index page question feed

This commit will fix mobile responsiveness
for index page question feeds container issue

Resolves: https://www.pivotaltracker.com/n/projects/2232110 [#163020832]

abidex4yemi committed:
updated UI index page for responsiveness

saheedt commented:
Looks good to merge.
